lib/core/src/Cardano/Wallet/Primitive/Types/UTxO.hs

```diff
@@ -212,6 +211,52 @@ partition f (UTxO u) = bimap UTxO UTxO $ Map.partitionWithKey (const . f) u
 toList :: UTxO -> [(TxIn, TxOut)]
 toList = Map.toList . unUTxO
 
+{-------------------------------------------------------------------------------
+    Delta encodings of UTxO
+-------------------------------------------------------------------------------}
+-- | Efficient delta encoding for 'UTxO'.
+data DeltaUTxO = DeltaUTxO
+    { excluded :: !(Set TxIn) -- ^ First exclude these inputs
+    , received :: !UTxO       -- ^ Then receive these additional outputs.
```

Comment:
Hi @HeinrichApfelmus 

Just wondering, why not remove a level of nesting:
```suggestion
    , received :: !(Map TxIn TxOut)       -- ^ Then receive these additional outputs.
```
My (hand-waving) justification is that this makes it obvious to the reader that the type of keys in `excluded` and the type of values in `received` are in fact the same type.

(Apologies for the drive-by comment!)

Reply:
Well, I would argue that within the scope of the `Cardano.Wallet.Primitive.Types.UTxO` module, the fact that the key type of `UTxO` is also `TxIn` is already "in scope" — no need to put an extra highlight on it.

By using the type `UTxO`, we can reuse existing concepts within the module, e.g. `excluding` in the definition of the `Semigroup` instance, or in the definition of `excludingD`.

The fact that `excludingS` and `restrictedByS` are defined in terms of the `Map` is a matter of performance, not of semantics. A semantic definition of `excludingS` would have been

    excludingS ins utxo = ins `Set.difference` dom utxo

but this definition allocates and destroys an extra `Set`.